packages/engine-test-utils/src/__tests__/engine-server/markdown/blockAnchors.spec.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  DendronASTData,
+  DendronASTDest,
+  DEngineClient,
+  MDUtilsV4,
+  UnistNode,
+  BlockAnchor,
+  blockAnchors,
+  BlockAnchorOpts,
+  DendronASTNode,
+} from "@dendronhq/engine-server";
+import _ from "lodash";
+
+function proc(
+  engine: DEngineClient,
+  dendron: DendronASTData,
+  opts?: BlockAnchorOpts
+) {
+  return MDUtilsV4.proc({ engine })
+    .data("dendron", dendron)
+    .use(blockAnchors, opts);
+}
+
+function genDendronData(opts?: Partial<DendronASTData>): DendronASTData {
+  return { ...opts } as any;
+}
+
+/** Gets the descendent (child, or child of child...) node of a given node.
+ *
+ * @param node The root node to start descending from.
+ * @param indices Left-to-right indexes for children, e.g. first index is for the root, second is for the child of the root...
+ * @returns Requested child. Note that this function has no way of checking types, so the child you get might not be of the right type.
+ */
+function getDescendentNode<Child extends DendronASTNode>(
+  node: UnistNode,
+  ...indices: number[]
+): Child {
+  const index = indices.shift();
+  if (_.isUndefined(index)) return node as Child;
+  expect(node).toHaveProperty("children");
+  expect(node.children).toHaveProperty("length");
+  // @ts-ignore
+  expect(node.children.length).toBeGreaterThanOrEqual(index);
+  // @ts-ignore
+  return getDescendentNode<Child>(node.children[index], ...indices);
+}
+
+function getBlockAnchor(node: UnistNode): BlockAnchor {
+  return getDescendentNode<BlockAnchor>(node, 0, 0);
+}
+
+describe("blockAnchors", () => {
+  describe("parse", () => {
+    let engine: any;
+    let dendronData = {
+      fname: "placeholder.md",
+      dest: DendronASTDest.MD_REGULAR,
+    };
+
+    test("parses anchor by itself", () => {
+      const resp = proc(engine, genDendronData(dendronData)).parse(
```

Comment:
I would add the following test utility which can return a *proper* type. this way, you can pass in an `DendronASTNode` to `getDescendentNode` and avoid the `@ts-ignore` statements
```ts
function invokeDendronProc(proc: Processor, text: string): DendronASTNode {
  return proc.parse(text) as DendronASTNode
}
```